### ISSUE TITLE
Listing @noble/hashes as a depenency.

### ISSUE THREAD

**Issue by einaralex** [shazow/whatsabi#88]
I just wanted to start by thanking you for this fantastic package; I've been struggling to identify ERC-1271 proxies and finally found an efficient way to do it with your package.

Listing @noble/hashes as a peerDependency in your package requires your users to install it as a dependency in their application. Ethers, Safe and other packages do this.

**Comment by einaralex** [comment]
I just realized this might not be a problem with npm and pnpm; I think they install peer dependencies from other package dependencies. We use Yarn.

**Comment by einaralex** [comment]
I created a PR if you're interested.

**Comment by shazow** [comment]
@einaralex Thanks for the kind words, and for raising this!

I'm honestly not 100% clear on how peerDependencies are supposed to work, despite having read the docs a dozen times.

Here's what I'm trying to achieve, maybe you can help me figure this out?

- `ethers` (a regular dependency) contains `@noble/hashes` as a dependency. (Also incidentally, `viem` (if you have it installed` also contains `@noble/hashes` as a dependency, and basically every other provider library)
- I want to use whatever existing version of `@noble/hashes` you already have, either installed directly or indirectly, without accidentally installing yet another version as a parallel dependency.

Any idea what the best way to do this in a way that is yarn-compatible? My understanding is that yarn v2+ has peerDependencies support?

**Comment by einaralex** [comment]
Yeah haha, I noticed I have 6 different versions of noble/hashes in my yarn.lock from different web3 packages.
I don't think there is a good automated way to do it. Since you list ethers as a dependency, I'd probably just follow that version, since the user of your app will get that version of ethers, you'd be matching that.

I just noticed we are actually using v1 of Yarn, so it's maybe more of a me-problem :P
